AudiusProject/audius-client
# [AUD-985] Consolidate error flow

### Description

Consolidates existing error flow and improves it by
 - Ensuring errors thrown in redux, and errors caught by React `ErrorBoundarys` are all ultimately handled by the `handleError` saga.
 - Properly implementing the "/error" route with react-router, instead of manual pathname checks that were happening in the App. Additionally, the error route behavior was divergent between saga errors and `ErrorBoundary` errors. For saga errors, we would push to "/error" route, and in app render, check pathname and conditionally render error page, whereas for ErrorBoundary errors, we would check if an error was logged to the state, and conditionally render error page (this would not change the pathname...).
 - Adds better ErrorBoundary component support (using `react-error-boundary` library), which can catch errors that occur in callbacks and other event handlers not inside redux. Before we essentially couldn't detect these errors!
 - Adds `react-router-last-location` to simplify getting our last location.

### Dragons

Things have worked as expected in my testing, but obviously with error handling we might get into weird edge-case states. I tried to keep functionality nearly the same for this pr. a follow up pr can make the "goBack" functionality use react-router instead of doing a full page refresh. This leads to much better performance as the page doesn't do a full refresh, but im worried it's too big of a change to also include here!

### Testing
On https://dylan.audius.co/ for testing. To trigger errors:
1. Go to /trending
2. Open console
3. Click the first track (should see "error from saga") this is a case where the "play action" puts a "handleError" action. This goes to the `handleError` saga
4. Click the second track (should see "error from ErrorBoundary"), this gets caught in the error boundary, which dispatches "handleError" action, which goes to the `handleError` saga

Confirm that in both cases, the pathname changes to /error, and we see the error page.
